Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy.swift

```diff
@@ -584,11 +584,18 @@ extension PathHierarchy {
                 let childNode = lookup[identifiers[child.nodeID]]!
                 // Even if this is a symbol node, explicitly pass the kind and hash disambiguation.
                 node.add(child: childNode, kind: child.kind, hash: child.hash)
+                if let kind = child.kind {
+                    childNode.symbol?.kind.identifier = .init(identifier: kind)
+                }
             }
         }
         
         self.lookup = lookup
-        self.modules = fileRepresentation.modules.mapValues({ lookup[identifiers[$0]]! })
+        let modules = fileRepresentation.modules.map({ lookup[identifiers[$0]]! })
+        for node in modules {
+            node.symbol?.kind.identifier = .module
+        }
+        self.modules = modules
```

Comment:
Is this necessary? Each node that is part of modules will have the .module as the identifier. I checked, and the test passes fine without this change.

Reply:
Yes, this is necessary. If I comment out the loop that sets the kind identifier of all modules I get 19 failures in `ExternalPathHierarchyResolverTests.testRedundantDisambiguations()`.

The reason why this is needed is that the symbols are created with empty kind identifiers and are being filled in while building up the hierarch, but the modules are top-level so they aren't processed while building up the hierarchy. We do it this way to make the file size smaller by minimizing repeated data.

I added more comments in https://github.com/apple/swift-docc/pull/770/commits/2b72f08c9f36405ee7dd3187a8a25218688d78b9 to describe this inline in the code.

Reply:
Thanks @d-ronnqvist, this is ready to merge 👍